diff --git a/crates/eww/src/widgets/widget_definitions.rs b/crates/eww/src/widgets/widget_definitions.rs
index 07a081d..7172f2a 100644
--- a/crates/eww/src/widgets/widget_definitions.rs
+++ b/crates/eww/src/widgets/widget_definitions.rs
@@ -927,6 +927,7 @@ const WIDGET_NAME_CALENDAR: &str = "calendar";
 /// @desc A widget that displays a calendar
 fn build_gtk_calendar(bargs: &mut BuilderArgs) -> Result<gtk::Calendar> {
     let gtk_widget = gtk::Calendar::new();
+
     def_widget!(bargs, _g, gtk_widget, {
         // @prop day - the selected day
         prop(day: as_f64) {
@@ -961,11 +962,20 @@ fn build_gtk_calendar(bargs: &mut BuilderArgs) -> Result<gtk::Calendar> {
                 run_command(
                     timeout,
                     &onclick,
-                    &[w.day(), w.month(), w.year()]
-                )
+                    &[w.day(), w.month() + 1, w.year()]
+                );
             }));
-        }
+        },
 
+        prop(timeout: as_duration = Duration::from_millis(200), onrightclick: as_string) {
+            connect_signal_handler!(gtk_widget, gtk_widget.connect_day_selected_double_click(move |w| {
+                run_command(
+                    timeout,
+                    &onrightclick,
+                    &[w.day(), w.month() + 1, w.year()]
+                );
+            }));
+        }
     });
 
     Ok(gtk_widget)
