diff --git a/crates/eww/src/widgets/widget_definitions.rs b/crates/eww/src/widgets/widget_definitions.rs
index 2cdc907..c01d904 100644
--- a/crates/eww/src/widgets/widget_definitions.rs
+++ b/crates/eww/src/widgets/widget_definitions.rs
@@ -920,6 +920,7 @@ const WIDGET_NAME_CALENDAR: &str = "calendar";
 /// @desc A widget that displays a calendar
 fn build_gtk_calendar(bargs: &mut BuilderArgs) -> Result<gtk::Calendar> {
     let gtk_widget = gtk::Calendar::new();
+
     def_widget!(bargs, _g, gtk_widget, {
         // @prop day - the selected day
         prop(day: as_f64) {
@@ -950,15 +951,14 @@ fn build_gtk_calendar(bargs: &mut BuilderArgs) -> Result<gtk::Calendar> {
         // @prop onclick - command to run when the user selects a date. The `{0}` placeholder will be replaced by the selected day, `{1}` will be replaced by the month, and `{2}` by the year.
         // @prop timeout - timeout of the command. Default: "200ms"
         prop(timeout: as_duration = Duration::from_millis(200), onclick: as_string) {
-            connect_signal_handler!(gtk_widget, gtk_widget.connect_day_selected(move |w| {
+            connect_signal_handler!(gtk_widget, gtk_widget.connect_day_selected_double_click(move |w| {
                 run_command(
                     timeout,
                     &onclick,
-                    &[w.day(), w.month(), w.year()]
-                )
+                    &[w.day(), w.month() + 1, w.year()]
+                );
             }));
         }
-
     });
 
     Ok(gtk_widget)
